src/NavBar.ts

```diff
@@ -95,26 +95,28 @@ export default class NavBar {
         const bbox = this.getBBoxFromUrlPoints(parseResult.points.map(p => p.coordinate))
         if (bbox) Dispatcher.dispatch(new SetInitialBBox(bbox))
 
-        // remove old query points
-        this.queryStore.state.queryPoints.forEach(point => Dispatcher.dispatch(new RemovePoint(point)))
-
-        // add parsed points
-        parseResult.points.forEach((point, i) => Dispatcher.dispatch(new AddPoint(i, point.coordinate, true)))
-
-        // assuming that at least two points should be present add un-initialized points if necessary
-        for (let i = this.queryStore.state.queryPoints.length; i < 2; i++) {
-            Dispatcher.dispatch(new AddPoint(i, { lat: 0, lng: 0 }, false))
-        }
-
-        // add routing profile
-        Dispatcher.dispatch(new SetVehicleProfile(parseResult.profile))
+        // we want either all the points replaced from the url or we just have one and we want to replace the one default
+        // one
+        const points = parseResult.points.length > 2 ? parseResult.points : this.fillPoints(parseResult.points)
```

Comment:
When do we have only a single point and with which default one do we want to replace this?

Reply:
This is the case, when the url contains only 1 point. Since we don't have information about the index of this point, this assumes that this single point is the starting point.

Reply:
Ah ok if I enter only one point in the url the first input field is populated and the second remains empty. So this seems to work :)

Reply:
Yes, that's what I meant :-)